src/bench/CMakeLists.txt

```diff
@@ -46,24 +47,23 @@ add_executable(bench_bitcoin
   strencodings.cpp
   util_time.cpp
   verify_script.cpp
+  $<$<BOOL:${ENABLE_WALLET}>:wallet_balance.cpp>
+  $<$<BOOL:${ENABLE_WALLET}>:wallet_create_tx.cpp>
+  $<$<BOOL:${ENABLE_WALLET}>:wallet_loading.cpp>
   xor.cpp
 )
 
 target_link_libraries(bench_bitcoin
   core
   test_util
-  leveldb
+  bitcoin_node
+  $<TARGET_NAME_IF_EXISTS:bitcoin_wallet>
+  bitcoin_common
+  bitcoin_consensus
+  bitcoin_util
+  bitcoin_crypto
+  $<TARGET_NAME_IF_EXISTS:bitcoin_zmq>
   univalue
+  leveldb
   Boost::headers
 )
-
```

Comment:
> This seems much less straightforward to me than before. It's also using a side-effect (TARGET_NAME_IF_EXISTS) rather than the config option directly.
> 
> Why the change? It also makes it very hard to tell what's actually changing.
> 
> So.. I suspect I'm missing some obvious reason why this is being refactored? Did we agree somewhere to standardize on this syntax?

In the following code:
```cmake
target_link_libraries(test_bitcoin
  core
  test_util
  bitcoin_node
  $<TARGET_NAME_IF_EXISTS:bitcoin_wallet>
  bitcoin_consensus
  bitcoin_common
  bitcoin_cli
  bitcoin_util
  bitcoin_crypto
  $<TARGET_NAME_IF_EXISTS:bitcoin_zmq>
  univalue
  minisketch
  leveldb
  libevent::libevent
  Boost::headers
)
```

using of the `TARGET_NAME_IF_EXISTS` generator expressions (which is one of the coolest CMake features) guarantees the needed linker order. For example, the `bitcoin_wallet` will be linked after `bitcoin_node` but before `bitcoin_consensus`.

When using an `if(ENABLE_WALLET)` block, the `bitcoin_wallet` can be linked only before or after the libs mentioned in the `target_link_libraries` commnad.

Reply:
Aha, got it thanks. That's a good enough reason to prefer `TARGET_NAME_IF_EXISTS` over the `if(foo)` syntax anywhere it's possible, I guess.